chain/pruned_block_dispatcher_test.go

```diff
@@ -406,6 +421,38 @@ func (h *prunedBlockDispatcherHarness) assertPeerReplied(
 	}
 }
 
+// assertPeerReplied asserts that the query peer replies with a block the
+// PrunedBlockDispatcher queried for.
+func (h *prunedBlockDispatcherHarness) assertPeerFailed(
+	blockChan <-chan *wire.MsgBlock, errChan, cancelChan <-chan error,
+	expectedErr error) {
+
+	h.t.Helper()
+
+	select {
+	case <-blockChan:
+		h.t.Fatalf("expected no reply from peer")
+
+	case err := <-errChan:
+		require.ErrorIs(h.t, err, expectedErr)
+		for _, hash := range h.hashes {
+			h.dispatcher.CancelRequest(*hash, err)
```

Comment:
Feels weird that we are re-implementing the logic in `GetBlock`, but I see there's no unit tests for `GetBlock` so nvm, we need to do more work here to patch tests in future PRs tho.

Reply:
Agree fully, this felt not right to me as well. Will write an issue to move the block cache into `btcwallet` and enhance the test coverage especially `GetBlock`

Reply:
https://github.com/btcsuite/btcwallet/issues/904